tools/reckless

```diff
@@ -0,0 +1,684 @@
+
+    def get_inst_details(self):
+        """
+        Populate installation details from a github repo url.
+        Return True if all data is found.
+        """
+        r = requests.get(self.git_url)
+        if r.status_code != 200:
+            return False
+        if 'git/tree' in self.git_url:
+            tree = r.json()['tree']
+        else:
+            tree = r.json()
+        entry_guesses = py_entry_guesses(self.name)
+        for g in entry_guesses:
+            for f in tree:
+                if f['path'] == g:
+                    self.entry = g
+                    break
+            if self.entry is not None:
+                break
+        if self.entry is None:
+            for g in unsupported_entry(self.name):
+                for f in tree:
+                    if f['path'] == g:
+                        # FIXME: This should be easier to implement
+                        print(f'entrypoint {g} is not yet supported')
+                        return False
+        dependency_info = ['requirements.txt', 'pyproject.toml']
+        for d in dependency_info:
+            for f in tree:
+                if f['path'] == d:
+                    self.deps = d
+                    break
+            if self.deps is not None:
+                break
+        if not self.entry:
+            return False
+        if not self.deps:
+            return False
+        return True
+
+
+def create_dir(r, directory):
+    """Creation of a directory at path `d` with a maximum new dir depth `r`"""
+    if os.path.exists(directory):
+        return True
+    elif r <= 0:
+        return False
+    elif create_dir(r-1, os.path.split(directory)[0]):
+        os.mkdir(directory, 0o777)
+        print(f'created directory {directory}')
+        if os.path.exists(directory):
+            return True
+
+
+def remove_dir(target):
+    try:
+        shutil.rmtree(target)
+        return True
+    except NotADirectoryError:
+        print(f"Tried to remove directory {target} that does not exist.")
+    except PermissionError:
+        print(f"Permission denied removing dir: {target}")
+    return False
+
+
+class Config():
+    """A generic class for procuring, reading and editing config files"""
+    def obtain_config(self, config_path, default_text, warn=False):
+        """Return a config file from the desired location. Create one with
+        default_text if it cannot be found."""
+        if isinstance(config_path, type(None)):
+            raise Exception("Generic config must be passed a config_path.")
+        assert isinstance(config_path, str)
+        # FIXME: warn if reckless dir exists, but conf not found
+        if os.path.exists(config_path):
+            with open(config_path, 'r+') as f:
+                config_content = f.readlines()
+            return config_content
+        print(f'config file not found: {config_path}')
+        if warn:
+            confirm = input('press [Y] to create one now.\n').upper() == 'Y'
```

Comment:
User interaction should most likely be in a function, and called in the command handlers, not buried in a class that one might call from other places too.